label-application-rule-config-service-api/src/main/proto/org/hypertrace/label/application/rule/config/service/v1/label_application_rule.proto

```diff
@@ -0,0 +1,96 @@
+syntax = "proto3";
+
+package org.hypertrace.label.application.rule.config.service.v1;
+
+option java_multiple_files = true;
+
+message LabelApplicationRule {
+  string id = 1;
+  LabelApplicationRuleData data = 2;
+}
+
+message LabelApplicationRuleData {
+  string name = 1;
+  MatchingCriteria matching_criteria = 2;
+  ApplicationCriteria application_criteria = 3;
+
+  message MatchingCriteria {
+    Condition condition = 1;
+
+    message Condition {
+      oneof condition {
+        LeafCondition leaf_condition = 1;
+        CompositeCondition composite_condition = 2;
+      }
+    }
+
+    message CompositeCondition {
+      LogicalOperator operator = 1;
+      repeated Condition children = 2;
+
+      enum LogicalOperator {
+        LOGICAL_OPERATOR_UNSPECIFIED = 0;
+        LOGICAL_OPERATOR_AND = 1;
+        LOGICAL_OPERATOR_OR = 2;
+      }
+    }
+
+    message LeafCondition {
+      string key = 1;
```

Comment:
We may want to specify this as a condition too based on the conversations. Adding regex support on the key here would require a breaking change, but something like below would support it.


```proto
    message LeafCondition {
      StringCondition key_condition = 1;
      oneof value_condition {
        StringCondition string_value_condition = 2;
        UnaryCondition unary_value_condition = 3;
        JsonCondition json_value_condition = 4;
      }
    }
```

Reply:
This makes sense. However do note that key in condition can be regex based, however for dynamic label we will require specific key. Isn't it?

Reply:
Given that the label has its own specification of source I don't think any labeling concerns are relevant to the conditions? Unless I'm misunderstanding?

Reply:
Not a concern. Just an observation.